src/webui/webapplication.cpp

```diff
@@ -234,13 +226,13 @@ void WebApplication::action_query_torrents()
     CHECK_URI(0);
     const RawStringMap& gets = request().gets;
 
-    print(btjson::getTorrents( toUtf8(gets["filter"]),
-                               toUtf8(gets["category"]),
-                               toUtf8(gets["sort"]),
-                               toUtf8(gets["reverse"]) == "true",
-                               toUtf8(gets["limit"]).toInt(),
-                               toUtf8(gets["offset"]).toInt()),
-           Http::CONTENT_TYPE_JSON);
+    print(btjson::getTorrents(gets["filter"],
+                              QString::fromUtf8(gets["category"].constData(), gets["category"].length()),
```

Comment:
In the correct UTF-8 string null byte means end of string and can not appear as a part of another character, and that is one of the reasons why UTF-8 is so popular.